pkg/operator/controller/dns/controller.go

```diff
@@ -433,6 +473,42 @@ func conditionChanged(a, b iov1.DNSZoneCondition) bool {
 	return a.Status != b.Status || a.Reason != b.Reason || a.Message != b.Message
 }
 
+// sanitizeRecordStatusCondition purges deprecated fields from DNS Record
+// status. Removes "Failed" (DNSRecordFailedConditionType) and replaces with
+// "Published" (DNSRecordPublishedConditionType). Returns updated condition
+// list.
```

Comment:
Is there a time in which this function can be removed? Say a OCP release? Could we include instructions on when this can be removed?

Reply:
Added to godoc.